packages/server/src/dataEntities/appointment.ts

```diff
@@ -145,7 +148,30 @@ export class Appointment {
         public readonly customerSig: string
     ) {}
 
-    public static fromIAppointment(appointment: IAppointment): Appointment {
+    public serialise(): SerialisedAppointment {
+        return {
+            _type: Appointment.TYPE,
+            contractAddress: this.contractAddress,
+            customerAddress: this.customerAddress,
+            startBlock: this.startBlock,
+            endBlock: this.endBlock,
+            challengePeriod: this.challengePeriod,
+            customerChosenId: this.customerChosenId,
+            nonce: this.nonce,
+            data: this.data,
+            refund: this.refund.toHexString(),
+            gasLimit: this.gasLimit,
+            mode: this.mode,
+            eventAddress: this.eventAddress,
+            topics: this.topics,
+            preCondition: this.preCondition,
+            postCondition: this.postCondition,
+            paymentHash: this.paymentHash,
+            customerSig: this.customerSig
+        }
+    }
+
+    public static deserialise(appointment: SerialisedAppointment) {
```

Comment:
Should check the value of _type in here?

Reply:
Added a check, it won't hurt!
Did the same for SerialisableBigNumber.